### PR TITLE
Add Bugs 3H format option

### PR DESCRIPTION
**Pull request by goebish**
Add Bugs 3H format option to Bugs3Mini protocol

Channel 10: Altitude hold

### CODE REVIEW

File: src/protocol/bugs3mini_nrf24l01.c [hunk 1]
```diff
@@ -228,9 +235,12 @@ static void send_packet(u8 bind)
                   | GET_FLAG(CHANNEL_VIDEO, FLAG_VIDEO)
                   | GET_FLAG(CHANNEL_FLIP, FLAG_FLIP);
         packet[13] = arm_flags
-                   | GET_FLAG(CHANNEL_LED, FLAG_LED)
-                   | GET_FLAG(CHANNEL_ANGLE, FLAG_ANGLE);
-        
+                   | GET_FLAG(CHANNEL_LED, FLAG_LED);
+        if(Model.proto_opts[PROTOOPTS_FORMAT] == FORMAT_MINI)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
239:  Missing space before ( in if(  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 2]
```diff
@@ -321,12 +331,22 @@ static void update_telemetry() {
...
-        if(packet[11] & 0x80)
-            Telemetry.value[TELEM_FRSKY_VOLT1] = 840; // Ok
-        else if(packet[11] & 0x40)
-            Telemetry.value[TELEM_FRSKY_VOLT1] = 710; // Warning
-        else
-            Telemetry.value[TELEM_FRSKY_VOLT1] = 640; // Critical
+        if(Model.proto_opts[PROTOOPTS_FORMAT] == FORMAT_MINI) {
+            if(packet[11] & 0x80)
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 840;  // Ok
+            else if(packet[11] & 0x40)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
334:  Missing space before ( in if(  
335:  Missing space before ( in if(  
337:  Missing space before ( in if(  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 3]
```diff
@@ -321,12 +331,22 @@ static void update_telemetry() {
...
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 710;  // Warning
+            else
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 640;  // Critical
+        }
+        else {  // 3H
+            if (packet[11] & 0x40)
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 710; // Warning
+            else if (packet[11] & 0x80)
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 640; // Critical
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
342:  If an else has a brace on one side, it should have it on both  
344:  At least two spaces is best between code and comments  
346:  At least two spaces is best between code and comments  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 4]
```diff
@@ -321,12 +331,22 @@ static void update_telemetry() {
...
+            else
+                Telemetry.value[TELEM_FRSKY_VOLT1] = 840; // Ok
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
348:  At least two spaces is best between code and comments  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 5]
```diff
@@ -268,41 +278,41 @@ static void make_address()
         0x2d9e, 0x95a4, 0x9c5c, 0xb4a6, 0xa9ce, 0x562b, 0x3e73, 0xb895, 0x6a82, 0x9437, 0x3d5a,
         0x4bb2, 0x6949, 0xc224, 0x6b3d, 0x23c6, 0x9ea3, 0xa498, 0x5c9e, 0xa652, 0xce76, 0x2b4b, 0x733a };
     
-    if(rxid_high==0x00 || rxid_high==0xFF)
+    if (rxid_high==0x00 || rxid_high==0xFF)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
281:  Missing spaces around ==  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 6]
```diff
@@ -268,41 +278,41 @@ static void make_address()
...
         rx_tx_addr[0]=0x52;
     else
         rx_tx_addr[0]=rxid_high;
     
     rx_tx_addr[1]=tx_hash;
     
-    if(rxid_low==0x00 || rxid_low==0xFF)
+    if (rxid_low==0x00 || rxid_low==0xFF)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
288:  Missing spaces around ==  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 7]
```diff
@@ -268,41 +278,41 @@ static void make_address()
...
         rx_tx_addr[2]=0x66;
     else
         rx_tx_addr[2]=rxid_low;
     
-    for(u8 end_idx=0;end_idx<23;end_idx++)
+    for (u8 end_idx=0; end_idx<23; end_idx++)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
293:  Missing spaces around <  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 8]
```diff
@@ -268,41 +278,41 @@ static void make_address()
...
         else
             start=(end_idx-7)*16+7;
         //calculate sequence length
-        if(end_idx>6)
+        if (end_idx>6)
         {
-            if(end_idx>15)
-                length=(23-end_idx)<<1;
+            if (end_idx>15)
+                length = (23-end_idx)<<1;
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
301:  Missing spaces around >  
303:  Missing spaces around >  
304:  Missing spaces around <<  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 9]
```diff
@@ -268,41 +278,41 @@ static void make_address()
...
             else
-                length=16;
+                length = 16;
         }
         else
-            length=(end_idx+1)<<1;
+            length = (end_idx+1)<<1;
         //scan for a possible solution using the current end
-        for(u8 i=0;i<length;i++)
+        for (u8 i=0;i<length;i++)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
309:  Missing spaces around <<  
311:  Missing spaces around <  
311:  Missing space after ;  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 10]
```diff
@@ -268,41 +278,41 @@ static void make_address()
...
         {
             index=(i>>1)*7+(((i+1)>>1)<<3);
             index=start+index-rxid_high;
-            if(index==rxid_low)
+            if (index==rxid_low)
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
315:  Missing spaces around ==  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 11]
```diff
@@ -315,27 +325,38 @@ static void make_address()
 
 static void update_telemetry() {
     u8 checksum = 0x6d;
-    for(u8 i=1; i<12; i++) {
+    for (u8 i=1; i<12; i++) {
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
328:  Missing spaces around <  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 12]
```diff
@@ -315,27 +325,38 @@ static void make_address()
...
     }
-    if(packet[0] == checksum) {
+    if (packet[0] == checksum) {
         Telemetry.value[TELEM_FRSKY_RSSI] = packet[3];
         TELEMETRY_SetUpdated(TELEM_FRSKY_RSSI);
-        if(packet[11] & 0x80)
-            Telemetry.value[TELEM_FRSKY_VOLT1] = 840; // Ok
-        else if(packet[11] & 0x40)
-            Telemetry.value[TELEM_FRSKY_VOLT1] = 710; // Warning
+        if(Model.proto_opts[PROTOOPTS_FORMAT] == FORMAT_MINI) {
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
334:  Missing space before ( in if(  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 13]
```diff
@@ -315,27 +325,38 @@ static void make_address()
...
     }
 }
 
 static u16 bugs3mini_callback()
 {
-    switch(phase) {
+    switch (phase) {
         case BIND1:
-            if( NRF24L01_ReadReg(NRF24L01_07_STATUS) & BV(NRF24L01_07_RX_DR)) { // RX fifo data ready
+            if ( NRF24L01_ReadReg(NRF24L01_07_STATUS) & BV(NRF24L01_07_RX_DR)) { // RX fifo data ready
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
359:  At least two spaces is best between code and comments  
359:  Mismatching spaces inside () in if  
```

File: src/protocol/bugs3mini_nrf24l01.c [hunk 14]
```diff
@@ -362,7 +383,7 @@ static u16 bugs3mini_callback()
             phase = BIND1;
             return PACKET_INTERVAL - WRITE_WAIT;
         case DATA1:
-            if( NRF24L01_ReadReg(NRF24L01_07_STATUS) & BV(NRF24L01_07_RX_DR)) { // RX fifo data ready
+            if ( NRF24L01_ReadReg(NRF24L01_07_STATUS) & BV(NRF24L01_07_RX_DR)) { // RX fifo data ready
```
Comment (PhracturedBlue) [inline src/protocol/bugs3mini_nrf24l01.c]: Lint Error:
```
386:  At least two spaces is best between code and comments  
386:  Mismatching spaces inside () in if  
```

**Comment by TheRealMoeder** [comment]
Please update Changelog

**Comment by goebish** [comment]
Ok, but I don't really like this idea, that will end up in merge conflicts if each PR updates it ...

**Comment by goebish** [comment]
... and here we are :p

wouldn't it be better to just edit it directly on github when required ?

**Comment by TheRealMoeder** [comment]
Yes, with the small number of developers that should be sufficient.

**Pull request merged**